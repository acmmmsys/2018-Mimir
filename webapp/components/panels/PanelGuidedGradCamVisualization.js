import React from 'react';
import { connect } from 'react-redux';
import { compose, withProps, nest, lifecycle } from 'recompose';

import Panel from 'layout/Panel';
import ShowcaseImage from 'components/ui/ShowcaseImage';

const enhance = compose(
    connect(
        state => ({
            guidedGradCam: state.cnn.guidedGradCamImages[`${state.cnn.selectedImageId}-${state.cnn.selectedLayer}-${state.cnn.selectedClass}`],
            loading: state.loading.guidedGradCam[`${state.cnn.selectedImageId}-${state.cnn.selectedLayer}-${state.cnn.selectedClass}`],
        })
    ),
    withProps(props => ({
        emptyMessage: 'A gradcam representation of the image will appear here',
        source: props.guidedGradCam ? 'data:image/jpeg;base64,' + props.guidedGradCam : '',
        isEmpty: !props.guidedGradCam
    })),
);

export default enhance(
    nest(Panel, ShowcaseImage)
);