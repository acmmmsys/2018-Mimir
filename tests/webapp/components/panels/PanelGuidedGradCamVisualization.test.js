import React from 'react';
import PanelGuidedGradCamVisualization from '../../../../webapp/components/panels/PanelGuidedGradCamVisualization'
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store'

const mockStore = configureStore();

let store;
let wrapper;

test('PanelGuidedGradCamVisualization does render', () => {
  const wrapper = shallow(
      <PanelGuidedGradCamVisualization store={mockStore(initialState)}>This is a test!</PanelGuidedGradCamVisualization>
  );
  expect(wrapper).toMatchSnapshot();
});
