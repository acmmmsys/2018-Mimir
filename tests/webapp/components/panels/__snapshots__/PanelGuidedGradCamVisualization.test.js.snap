// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelGuidedGradCamVisualization does render 1`] = `
<withProps(nest(defaultProps(branch(branch(Panel))), ShowcaseImage))
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
>
  This is a test!
</withProps(nest(defaultProps(branch(branch(Panel))), ShowcaseImage))>
`;
